python/ray/_private/resource_spec.py

```diff
@@ -218,6 +218,7 @@ def resolve(self, is_head: bool, node_ip_address: Optional[str] = None):
                     resources[
                         f"{ray_constants.RESOURCE_CONSTRAINT_PREFIX}{accelerator_type}"
                     ] = 1
+                accelerator_manager.postprocess_resources(resources=resources)
```

Comment:
Nit: I think it's better/safer to have a function that returns additional resources and get added here than pass in and modify the entire resources dict.

Maybe like `resources.add(accelerator_manager.get_current_node_additional_resources())`

Reply:
I think so too, but unfortunately I have to remove the "tpu accelerator type" resource from the non worker 0 workers to get the correct intended behavior.

I can't think of a way to avoid this since I have to assign the TPU accelerator type resource at the cluster launcher/autoscaler level which will propagate this resource to all workers in the pod slice.

Reply:
Oh, I didn't realize there is such interaction. Could you add more comments to describe this? It's tricky.

Can we change node provider and TPUCommandRunner to only set this resource for worker 0?

Reply:
Yes, I added in this comment in the GCP node provider where this happens (in the fillout). 

I have previously tried to do this same PR in just the node provider and TPUCommandRunner (see this draft PR: https://github.com/ray-project/ray/pull/41065/files) but I didn't like it because it wouldn't work for Kuberay. Do you think I should apply some of that concept here as well?

Reply:
Done